common/persistence/sql/storage/sqldb/interfaces.go

```diff
@@ -681,21 +688,51 @@ type (
 		GetAckLevels(queueType common.QueueType, forUpdate bool) (map[string]int, error)
 	}
 
+	// AdminCURD defines admin operations for CLI and test suites
+	AdminCURD interface {
+		CreateSchemaVersionTables() error
+		ReadSchemaVersion(database string) (string, error)
+		UpdateSchemaVersion(database string, newVersion string, minCompatibleVersion string) error
+		WriteSchemaUpdateLog(oldVersion string, newVersion string, manifestMD5 string, desc string) error
+		ListTables(database string) ([]string, error)
+		DropTable(table string) error
+		DropAllTables(database string) error
+		CreateDatabase(database string) error
+		DropDatabase(database string) error
+		Exec(stmt string, args ...interface{}) error
+	}
+
 	// Tx defines the API for a SQL transaction
 	Tx interface {
 		tableCRUD
 		Commit() error
 		Rollback() error
 	}
 
-	// Interface defines the API for a SQL database
-	Interface interface {
+	// DB defines the API for SQL operations
+	DB interface {
 		tableCRUD
+		AdminCURD
+
 		BeginTx() (Tx, error)
 		DriverName() string
+		IsDupEntryError(err error) bool
 		Close() error
 	}
 
+	// DBAndTx is mainly for fixing the Lint warning and also simplify the type assertion
```

Comment:
Yes two assertion is fine. The actual problem is lint warning as we always want to clear them. So if the return type of Init() is db struct, then the warning will arise because db is a private member. If it is DB interface then it worn't work(compile error) for beginTx()...

I agree that this is a little bit confusing. So here are all the options:
1. Make db struct public, maybe call it MySQLDB/PostgresDB -- but all member fileds are private. This is against of design pattern because we want to hide the implementation details.
2. Returning db, ignore the lint warning. 
3. Suppress lint warning.

I can't find a way to do 3 yet. So maybe 2 is our best option.

Reply:
I have changed it to option 2. Let's see if @venkat1109 have any other opinions.